annlite/executor.py

```diff
@@ -303,6 +298,5 @@ def close(self, **kwargs):
             time.sleep(0.1)
 
         with self._index_lock:
-            self._data_buffer = None
-            self._index_thread.join()
-            self._index.close()
+            super().close()
+            del self._index
```

Comment:
Please double check `del self._index` will not erase the file content located at the `data_path`